File: no-require-server/CoderByte Challenges/JavaScript/challenge_4.js
//strManipulator
//manipulate letters in strings according to their position in alphabet and their case

function LetterChanges(str) {
    var letters = /^[A-Za-z]+$/;
    var strArr = [];
    var charCodeArr = [];
    var vowels = ["a", "e", "i", "o", "u"];
    str = str.toLowerCase();
    for (var counter = 0; counter < str.length; counter++) {
        strArr[counter] = str[counter];
        if (strArr[counter] == "z") {
            charCodeArr[counter] = 65;
        }
        else if (strArr[counter].match(letters)) {
            charCodeArr[counter] = (str.charCodeAt(counter) + 1);
        }
        else {
            charCodeArr[counter] = str.charCodeAt(counter);
        }
    }
    for (counter = 0; counter < str.length; counter++) {
        strArr[counter] = String.fromCharCode(charCodeArr[counter]);
    }
    for (counter = 0; counter < str.length; counter++) {
        str[counter] = strArr[counter]
    }
    str = "";
    for (counter = 0; counter < strArr.length; counter++) {
        str += strArr[counter]
    }
    str = "";
    for (counter = 0; counter < strArr.length; counter++) {
        if (vowels.includes(strArr[counter])) {
            str += strArr[counter].toUpperCase();
        }
        else {
            str += strArr[counter]
        }
    }
    return str;
}
LetterChanges(readline());

File: no-require-server/CoderByte Challenges/Python/challenge_4.py
# strManipulator
# manipulate letters in strings according to their position in alphabet and their case
def LetterChanges(str):
    firstCounter = 0;
    secondCounter = 0;
    thridCounter = 0;
    str = list(str);
    for letter in str:
        if(letter.isalpha()):
            if(letter.lower() != 'z'):
                str[firstCounter] = chr(ord(letter) + 1).lower();
            else: str[firstCounter] = 'a';
        else:pass;
        firstCounter += 1;
    str = "".join(str);
    str = list(str);
    for letter in str:
       if(letter.lower() in ['a', 'e', 'i', 'o', 'u']):
            str[secondCounter] = letter.upper();
       secondCounter += 1;
    str = "".join(str);
    return str

while True:
    user_input = input("Type any string to mess the shit out of it: ")
    print(LetterChanges(user_input));
